### PR TITLE
Add config for getGPUTier method in cornerstone.init

### PR DESCRIPTION
**Pull request by iamiota**
### Context
When cornerstone3D is being initialized, the `getGPUTier` method from `detect-gpu` will generate a network request, which will not be implemented within a LAN and at the same time will block the initialization of cornerstone3D.

Move the parameter of `getGPUTier` to the parameter of `cornerstone3D.init`, users can configure the path of device information by themselves. For example, they can download the device information files to a specific path on the hard disk in advance, so that even in a LAN, files can still be requested.

Perhaps the name `getGPUTierConfig` is not accurate enough, do you have any suggestions?

### Changes & Results
```typescript
import * as cornerstone from '@cornerstonejs/core'
cornerstone.init({
  getGPUTierConfig: {
    benchmarksURL: '/resources/devices',
  },
})
```
#### Before
WX20230526-145648@2x

#### After
WX20230526-145712@2x

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

- [] I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS:  macOS 12.6.2
- [] "Node version:  16.13.0
- [] "Browser: Chrome 113.0.5672.126
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

**Comment by sedghi** [comment]
Hey, I really like the idea. But I don't see the implementation would work. I think this is what you are trying to do

the init function would be used like this 

`cornerstone3D.init({getGPUConfig: ()=>{}})` 

then inside our `init` you should check if the function is provided and use that instead of the network request and bypass the detect-gpu. 

Let me know what you think. 
Also we should update the docs to reflect this change after

**Comment by iamiota** [comment]
I'm sorry, I didn't express myself clearly.
I don't want to generate a network request to replace `getGPUTier`.

First, I looked at the `detect-gpu` documentation and found out that the getGPUTier method allows an object parameter to be passed in, including a property called [benchmarksURL](https://github.com/pmndrs/detect-gpu#:~:text=benchmarksURL?:%20string;), which overrides the default device information request address (the default address is the request address shown in the first screenshot). 

So what I want to do is pass the benchmarksURL property to getGPUTier through the cornerstone init method. The code I submitted is effective, and the screenshot above is the result of my local verification.

I don't know if it's elegant to pass like this in the `cornerstone.init`, please forgive my poor English.
Thanks!

File: common/reviews/api/core.api.md [hunk 1]
```diff
@@ -210,6 +211,7 @@ enum ContourType {
 // @public (undocumented)
 type Cornerstone3DConfig = {
     detectGPU: any;
```
Comment (sbsurf) [inline common/reviews/api/core.api.md:213]: Wouldn't passing in an instance of `getGPUTier` to `detectGPU` with the `benchmarksURL` option set work?

**Comment by sedghi** [comment]
@iamiota I see ok makes sense. Looking back at this `detectGPU` does not makes sense, it should be renamed to `gpuTier` since it holds the tier, and you can add the `detectGPUConfig` and add your benchmark url

**Comment by iamiota** [comment]
After careful consideration, I have resubmitted some code. Overall, the description of this PR is as follows:

1. Renamed `detectGPU` to `gpuTier` and specified its type as [`TierResult`](https://github.com/pmndrs/detect-gpu/blob/master/src/index.ts#L82)(which is the return value of `getGPUTier`), it is `optional`. Users could obtain device information in their own way and pass it as a parameter to `cornerstone3D.init`. When this value exists, `cornerstone3D.init` method will use this parameter instead of making a network request to retrieve device information.

2. Added the `detectGPUConfig` parameter, which is of type [`GetGPUTier`](https://github.com/pmndrs/detect-gpu/blob/master/src/index.ts#L20) (the parameter for `getGPUTier`). This allows users to configure it based on their network conditions.

Thanks! @sedghi  @sbsurf

File: packages/core/src/types/Cornerstone3DConfig.ts [hunk 2]
```diff
@@ -1,5 +1,8 @@
+import type { TierResult, GetGPUTier } from 'detect-gpu';
+
 type Cornerstone3DConfig = {
-  detectGPU: any;
+  gpuTier?: TierResult;
```
[the remark this answers is not in the document]
Comment (iamiota) [reply packages/core/src/types/Cornerstone3DConfig.ts:10]: Of course, it has already been added.

**Comment by sedghi** [comment]
You should run `yarn build:update-api` and include(commit) the results. I have done that for you this time, just FYI for next time

**Review by sedghi — APPROVED** [review]

**Pull request merged**